HACKING.md

```diff
@@ -1,5 +1,127 @@
-# Hacking on the `cluster-logging-operator`
+# Hacking on the Cluster Logging Operator
+
+## Preparation
+
+Clone this repository into `$GOPATH/github.com/openshift/cluster-logging-operator`
+
+Install `podman`. For example on  Fedora:
+```
+sudo dnf install -y podman
+```
+
+Check out some other required repositories in your $GOPATH:
+```
+go get -d -u github.com/openshift/elasticsearch-operator/... github.com/openshift/origin-aggregated-logging/...
+```
+
+**Note:** `podman` copies our source tree very slowly. You can use `docker`
+instead, this will not exactly reproduce how images are built for release or by
+CI, but is close enough for many purposes. Put a symlink early in your path:
+```
+ln -s /usr/bin/podman /somewhere/early/in/my/PATH`
+```
+
+## Testing Overview
+
+There are two types of test:
+
+* *Unit tests:* run in seconds, no cluster needed, verify internal behavior - behind the API. New could *should* be reasonably well covered by unit tests but we don't have a formal requirement.
+* *E2E tests:* run in minutes to hour(s), requires a cluster. There are several ways to run these:
+   - *CI tests:* run automatically when you raise a PR in a controlled cluster environment. Can take hour(s) to get results.
+   - *Deployed tests:* Build and deploy CLO image to your own cluster, allows you to run tests selectively.
+   - *Local process:* requires a cluster with dependencies deployed (elasticsearch), but runs CLO as a *local process*. Faster and easier to debug.
+
+All PRs should pass unit tests. CI runs all tests automatically and won't merge a PR till they pass, so it is OK to create a PR and let CI run the e2e tests if you've done enough testing to be confident. This allows for human review and feedback while CI tests are running.
+
+To get a code review of unfinished work, create a PR with "[WIP]" at the start of the description. The CI system will not merge it until the "[WIP]" is removed and all tests are passing.
+
+## Setting up a test cluster
+
+You can use a real cluster (for example an AWS cluster) or create a virtual cluster on your development box with [Code Ready Containers](https://developers.redhat.com/products/codeready-containers/download)
+
+If you use CRC, you may need to start with more memory than the default, e.g.:
+```
+crc start -m 12288
+```
+
+Log in as a user with the role `cluster-admin` - user `kubeadmin` is often predefined with this role.
+
+You can `export KUBECONFIG=/path/to/cluster/config` or copy/add your config to `$HOME/.kube/config`. Depending on how you set it up, you may also need to `oc login` to your cluster and add pull secrets to your credentials.
+
+## Running tests
+
+Unit tests can be run without a cluster:
+* `make`: Run unit tests, lint and format code. Do this before submitting a PR.
+* `make test-unit`: Just run unit tests.
+
+To build, deploy and test the CLO image in your own cluster:
+* `make deploy`: Build CLO image and deploy to cluster with elasticsearch.
```

Comment:
does this do an OLM based install of the operator?

Reply:
Yes, @jcantrill rewrote the "deploy" and "undeploy" targets to use your targets, and your targets are still there if someone want's finer control.